src/rig.c

```diff
@@ -6219,17 +6214,12 @@ int HAMLIB_API rig_get_powerstat(RIG *rig, powerstat_t *status)
     HAMLIB_TRACE;
     retcode = rig->caps->get_powerstat(rig, status);
 
-    if (retcode == RIG_EIO)
+    if (retcode != RIG_OK)
     {
-        rig_debug(RIG_DEBUG_ERR, "%s: hard error, reopening rig\n", __func__);
-        rig_close(rig);
-        rig_open(rig);
+        // if failed, assume power is on
+        *status = RIG_POWER_ON;
     }
 
-    if (retcode != RIG_OK) { *status = RIG_POWER_ON; } // if failed assume power is on
-
-    if (*status == RIG_POWER_OFF && rig->state.auto_power_on) { rig->caps->set_powerstat(rig, RIG_POWER_ON); }
```

Comment:
`get_powerstat` should never alter power status IMO

Reply:
This is wrong....that was put in specifically for Flex.  It fails get_power_stat when power is off.  No way to tell if it's on so we have to assume it's on.  Same would go for any rig that fails that call.  If we cannot get power status from the rig then we simply have to assume it's on.  That would include using other calls like get_freq but it would have to be done in the rigs backend.  get_freq failing is one possible indication of rig power off, or bad comm but can't really tell the difference.  Bad comm would be the equivalent of power off -- or at least "rig not responding" which we don't have an error code for and not sure that's needed.

Reply:
Ok, so you're only talking about the first if statement.

I think it should be enough for the rig backend to say get_powerstat is not supported, just like most (all?) Icom rigs do.

Why would that not be an option with Flex?

With Icom rigs this is already how it works and there is no way to know whether a timeout means that the rig is powered off or if there is a communication problem.

Reply:
Additionally, there should be no need to "mask" rig error codes -- the client should know if get_powerstat fails for rigs that support it (in which case it would most likely be a comm failure)